tests/0011-produce_batch.c

```diff
@@ -106,7 +115,14 @@ static void test_single_partition(void) {
         TEST_SAY("test_single_partition: Created kafka instance %s\n",
                  rd_kafka_name(rk));
 
-        rkt = rd_kafka_topic_new(rk, test_mk_topic_name("0011", 0), topic_conf);
+        topicSuffix = (char *)malloc(topicSuffixLength + 1 * sizeof(char));
```

Comment:
This change can be removed as it's not necessary for reproducing the issue

Reply:
We need `topicSuffix` because with the combination of both `clientID` and `topicSuffix` we were able to see this border condition in the segment.

Reply:
I had verified it happens with just that client id, because the segment is removed when the varint is changed without reducing the erased bytes

Reply:
Removed suffix